bridge/svix-bridge/src/runtime/mod.rs

```diff
@@ -1,45 +1,128 @@
 use anyhow::Result;
-use std::sync::Arc;
-
-use deno_core::{anyhow, serde_v8, v8, JsRuntime};
+use deadpool::unmanaged::Pool;
+use deno_runtime::deno_core::{
+    serde_v8,
+    v8::{self, OwnedIsolate},
+    JsRealm, JsRuntime,
+};
 use svix_bridge_types::{JsObject, TransformerInput, TransformerOutput};
-use threadpool::ThreadPool;
-use tokio::sync::{oneshot, Mutex};
+use tokio::sync::oneshot;
+
+struct Executor {
+    tx: std::sync::mpsc::Sender<Job>,
+    _handle: std::thread::JoinHandle<()>,
+}
+
+impl Default for Executor {
+    fn default() -> Self {
+        let (tx, rx) = std::sync::mpsc::channel::<Job>();
+        let _handle = std::thread::spawn(move || {
+            let mut runtime = JsRuntime::new(Default::default());
+            // Realms expose a similar script execution API (no module support yet) as the Runtime itself,
+            // but takes an Isolate as an argument to scope the evaluation by.
+            // It _looks like_ you're supposed to be able to cheaply create a Realm and drop it to dispose
+            // of any heap associated with it, but this is not actually the case.
+            // For whatever reason, dropping a Realm leaks just like dropping a Runtime.
+            // As such, each Executor gets its own Runtime _and Realm_ which are each reused.
+            // Hopefully discarding the Isolate after each invocation is enough to prevent state from
+            // carrying over from call to call.
+            // Refs:
+            // - <https://github.com/denoland/deno/issues/17861>
+            // - <https://github.com/denoland/deno/issues/17199>
+            let realm = runtime.create_realm().expect("create realm");
+            for Job { input, script, cb } in rx {
+                let isolate = runtime.v8_isolate();
+                let ret = run_script_inner(isolate, &realm, input, script);
+                cb.send(ret).expect("callback");
+            }
```

Comment:
It's looking more and more like `JsRealm` won't help much for anything. The isolate fed in there from the runtime is always the same (belonging to the runtime), not a fresh one.

Constructing a new Isolate using the v8 APIs causes a panic here. May as well skip the realm entirely and just try to mitigate the global namespace from being polluted some other way.